src/main/java/DilemmaDetector/Simulator/SimulatorEngine.java

```diff
@@ -98,14 +101,27 @@ public SimulatorEngine(Model model, CollisionConsequencePredictor consequencePre
                         SIMULATION_TIME = updateSimulationTime(currentTime);
                     }
                 }
-                else if(!factoryWrapper.isPedestrian(actor)){
+                else if (factoryWrapper.isSurrounding(actor)){
+                    collisionWithSurrounding = true;
+                }
+                else if(factoryWrapper.isVehicle(actor)){
                     if (!actor.equals(mainVehicle)) {
-                        collisionNotWithPedestrian = true;
+                        collisionWithVehicle = true;
                     }
                 }
+                else if(factoryWrapper.isCollidableObstacle(actor.getEntityName())){
+                    collisionWithObstacle = true;
+                }
             }
+
+            Set<Actor> collidedInMomentWithoutSurroundingAndObstacles =
+                    collidedInMoment.stream().filter(
+                            a -> !factoryWrapper.isSurrounding(a) &&
+                                    !factoryWrapper.isCollidableObstacle(a.getEntityName())).
```

Comment:
To samo co wyżej. Ale trochę nie rozumiem co tu się dzieje, czemu to takie skomplikowane się zrobiło.

Reply:
Prawda, że jest trochę skomplikowane. Spowodowane jest to tym że wcześniej jak mieliśmy kolizję z surroundingiem to dodawałem do zbioru collided tylko main vehicle, przez co nie wiedziałem że to właśnie z surroundingiem się skolidowaliśmy. 
Wcześniej nie miało znaczenia z kim konkretnie się skolidowaliśmy tylko kolizja == koniec symulacji. Teraz mamy zmianę - gdy kolizja z pieszym to symulujemy jeszcze przez pewien czas, a gdy kolizja z surroundingiem/przeszkodą/pojazdem to kończymy. 
Dlatego dodaję w detectCollisionInMoment także surrounding. Gdy skończymy symulację muszę odfiltrować surroundingi i obstacles właśnie w tym streamie. 
Faktycznie można zmienić na isObstacle